packages/blitz-dom/src/events/driver.rs

```diff
@@ -55,8 +55,52 @@ impl<'doc, Handler: EventHandler> EventDriver<'doc, Handler> {
             UiEvent::MouseMove(event) => {
                 let dom_x = event.x + viewport_scroll.x as f32 / zoom;
                 let dom_y = event.y + viewport_scroll.y as f32 / zoom;
-                self.doc_mut().set_hover_to(dom_x, dom_y);
+                let changed = self.doc_mut().set_hover_to(dom_x, dom_y);
+
+                let old_chain = hover_node_id.map(|id| self.doc().node_chain(id));
+                let new_chain = self.doc().hover_node_id.map(|id| self.doc().node_chain(id));
+
+                // Find the difference in the node chain of the last hovered objected and the newest
+                let mut chain_diff = 0;
+                if let Some(ref old_chain) = old_chain {
+                    if let Some(ref new_chain) = new_chain {
+                        let old_len = old_chain.len();
+                        let new_len = new_chain.len();
+
+                        for i in 0..(old_len.min(new_len)) {
+                            if old_chain[old_len - i - 1] != new_chain[new_len - i - 1] {
+                                chain_diff = i;
+                            }
+                        }
+                    }
+                }
+
+                if changed {
+                    if let Some(target) = hover_node_id {
+                        self.handle_dom_event(DomEvent::new(target, DomEventData::MouseOut(event.clone())));
+
+                        // Send an mouseleave event to all old elements on the chain
+                        if let Some(ref old_chain) = old_chain {
+                            for i in chain_diff..old_chain.len() {
+                                self.handle_dom_event(DomEvent::new(old_chain[old_chain.len() - i - 1], DomEventData::MouseLeave(event.clone())));
+                            }
```

Comment:
This needs to handle the `chain_diff > old_chain.len()` case. I'm not sure that it currently does.

I would recommend iterating over `old_chain.get(chain_diff..).unwrap_or(&[]).iter().rev()`